raiden-ts/tests/unit/epics/transport.spec.ts

```diff
@@ -1477,4 +1478,259 @@ describe('transport epic', () => {
+        {
+          userId: partnerUserId,
+          available,
+          ts: Date.now(),
+          caps: { [Capabilities.WEBRTC]: webRtcCapable },
+        },
+        { address: partner },
+      );
+
+    let rtcConnection: ReturnType<typeof mockRTC>['rtcConnection'];
+    let rtcDataChannel: ReturnType<typeof mockRTC>['rtcDataChannel'];
+    let RTCPeerConnection: ReturnType<typeof mockRTC>['RTCPeerConnection'];
+    beforeEach(() => {
+      ({ rtcConnection, rtcDataChannel, RTCPeerConnection } = mockRTC());
+      action$.next(
+        raidenConfigUpdate({
+          httpTimeout: 300,
+          caps: { [Capabilities.NO_DELIVERY]: true, [Capabilities.WEBRTC]: true },
+        }),
+      );
+      matrix.getRoom.mockReturnValue({
+        roomId: partnerRoomId,
+        name: partnerRoomId,
+        getMember: jest.fn(() => ({
+          membership: 'join',
+          roomId: partnerRoomId,
+          userId: partnerUserId,
+        })),
+        getJoinedMembers: jest.fn(),
+        getCanonicalAlias: jest.fn(() => partnerRoomId),
+        getAliases: jest.fn(() => []),
+      } as any);
+    });
+    afterEach(() => {
+      RTCPeerConnection.mockRestore();
+    });
+
+    test('skip if no webrtc capability exists', async () => {
+      const promise = rtcConnectEpic(action$, state$, depsMock).toPromise();
+      action$.next(createPartnerPresence(false, false));
+      action$.next(createPartnerPresence(true, false));
+      setTimeout(() => action$.complete(), 100);
+
+      await expect(promise).resolves.toBeUndefined();
+    });
+
+    test('reset data channel if user goes offline in matrix', async () => {
+      const promise = rtcConnectEpic(action$, state$, depsMock).toPromise();
+      action$.next(createPartnerPresence(true, true));
+      action$.next(createPartnerPresence(false, true));
+      setTimeout(() => action$.complete(), 100);
+
+      await expect(promise).resolves.toEqual({
+        meta: { address: partner },
+        payload: undefined,
+        type: 'rtcChannel',
+      });
+    });
+
+    test('set up caller data channel and waits for partner to join room', async () => {
+      const promise = rtcConnectEpic(action$, state$, depsMock).toPromise();
+      action$.next(createPartnerPresence(true, true));
+      action$.next(matrixRoom({ roomId: userId }, { address: partner }));
+      action$.next(createPartnerPresence(false, true));
+      setTimeout(() => action$.complete(), 100);
+
+      await expect(promise).resolves.toEqual({
+        meta: { address: partner },
+        payload: undefined,
+        type: 'rtcChannel',
+      });
+    });
+
+    test('success callee, receive message & channel error', async () => {
+      expect.assertions(2);
+
+      // change partner to one with smaller address, to be the caller
+      partner = '0x0000000000000000000000000000000000000088' as Address;
+      partnerUserId = `@${partner.toLowerCase()}:${matrixServer}`;
+      matrix.getRoom.mockReturnValue({
```

Comment:
This looks the same as the one used above, is there any specific reason for the duplication? Could this be extracted to some method?

Reply:
It's because I changed the `partner` & `partnerUserId`, I needed to update the object to be returned by `getRoom` and `getMember`. I think it can be replaced by a `mockImplementation*` of some sort, will do it to simplify this part